docs/source/format/CanonicalExtensions.rst

```diff
@@ -283,6 +283,77 @@ UUID
    A specific UUID version is not required or guaranteed. This extension represents
    UUIDs as FixedSizeBinary(16) with big-endian notation and does not interpret the bytes in any way.
 
+Other
+=====
+
+Other represents a type or array that one Arrow-based system received from an
+external (likely non-Arrow) system, but cannot interpret itself.  In this
+case, the Other type explicitly communicates the name and presence of a field
+to downstream clients.
+
+For example:
+
+* A Flight SQL service may support connecting external databases.  In this
+  case, its catalog (``GetTables`` etc.) should reflect the names and types of
+  tables in external databases.  But those external systems may support types
+  it does not recognize.  Instead of erroring or silently dropping columns
+  from the catalog, it can use the Other[Null] type to report that a column
+  exists with a particular name and type name in the external database; this
+  lets clients know that a column exists, but is not supported.
+
+* The ADBC PostgreSQL driver, because of how the PostgreSQL wire protocol
+  works, may get bytes for a field whose type it does not recognize (say, a
+  geospatial type).  It can still return the bytes to the application which
+  may be able to parse the data itself.  In that case, it can use the
+  Other[binary] type to return the column data.  The Other type differentiates
+  the column from actual binary columns.
+
+Of course, the intermediate system *could* implement a custom extension type
+for these example types.  But there is no way in general that every type can
+be known in advance.  In such cases, the Other type allows the system to
+explicitly note that it does not support some type or field, without silently
+losing data or sending irrelevant errors.  It could also pretend to support
+the types by making up extension types on the fly.  But this misleads
+downstream systems who cannot tell if the type is supported or not.
+
+Extension parameters:
+
+* Extension name: ``arrow.other``.
+
+* The storage type of this extension is any type.  If there is no underlying
+  data, the storage type should be Null.  If there is data (because the system
+  got bytes or some other data it does not know how to interpret), the storage
+  type should preferably be binary or fixed-size binary, but may be any type.
+
+* Extension type parameters:
+
+  * **type_name** = the name of the unknown type in the external system.
+  * **vendor_name** = the name of the external system.
```

Comment:
Do we want to support an optional `mime_type` key or something?

Reply:
I would be in favor of that

Reply:
> I don't think that any of the places we envision getting this information have any notion of mime type

Who is "we"? Arrow is an open format, it's not company-specific. And you can put tons of different things in databases, not to mention that Arrow is not limited to representing database contents.

Reply:
mime_type seems overly restrictive and tunnel-visions on the original misleading suggestion in the mailing list of a JSON type...Also the use cases listed are not going to have a mime_type. 

The point again is to NOT provide another way to have generic extension types; this info is purely meant to communicate a bit of human-readable information for context and is NOT meant to provide a convention for applications to coalesce behind. If there's a MIME type you want to see supported, create an actual extension type for it.

Reply:
> mime_type seems overly restrictive and tunnel-visions on the original misleading suggestion in the mailing list of a JSON type...

How is it restrictive to _add_ information? Types which don't have a corresponding MIME type would simply not have that metadata.

(and what does JSON have to do with it?)

> Also the use cases listed are not going to have a mime_type.

Which "use cases listed"?

> If there's a MIME type you want to see supported, create an actual extension type for it.

Creating an extension type doesn't make it magically supported by third-parties.

Reply:
> The target audience here is (I think) general-purpose something-to-Arrow converters, and I am just not aware of any source of that "something" that provides a MIME type (e.g., if libpq had a GetColumnMIMEType function).

"Something"s don't produce MIME types because they produce actual types from their internal type system. It would be the converter's task to infer a MIME type from the "something type" when possible.

Reply:
The use cases are listed in the doc and have been listed this whole time.

The point of the type is that the converter doesn't know much about it. Asking it to infer a MIME type is just useless ceremony.

Reply:
But at the point where the converter can infer a MIME type, it actually knows the type, and then why not return this annotated with a normal (non-other) extension type?

Reply:
In any case: my broad point is that something like MIME type is almost an antipattern. This isn't meant to be a generic container for things with MIME types. This is meant for an intermediate Arrow system to declare that they are aware of a field but do not know its true type, and to communicate that. It is not about the consumer, and it is not meant as a base for a consumer to start developing conventions about particular representations.

Reply:
> The use cases are listed in the doc and have been listed this whole time.

Well, I thought this spec was more open-ended. You're basically saying "Other should only be used when a converter receives a type it did not know about in advance", right?

Reply:
Effectively, yes.

Reply:
>> You're basically saying "Other should only be used when a converter receives a type it did not know about in advance", right?
>
> Effectively, yes

(I still think that "Unknown" might signal that meaning better than "Other")

Reply:
This should probably be spelled out explicitly, and adding a suggestion to create bespoke extension types otherwise.

Reply:
The suggestion is already present